src/main/java/shop/service/impl/OrderServiceImpl.java

```diff
@@ -4,47 +4,57 @@
 import shop.dao.impl.OrderDAOImpl;
 
 import shop.entity.Order;
+import shop.entity.Product;
 import shop.service.OrderService;
 import shop.service.ProductService;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.UUID;
 
 import static shop.enums.OrderStatus.*;
 
 public class OrderServiceImpl implements OrderService {
+
     private OrderDAO orderDAO = new OrderDAOImpl();
     private ProductService productService = new ProductServiceImpl();
+    private List<Product> productCart = new ArrayList<>();
 
     @Override
-    public void addProductToOrder(String productId) {
-         productService.getProduct(productId);
-    }
-
-           @Override
-            public void getOrder(String orderId) {
+    public void getOrder(String orderId) {
 
-            }
+    }
 
-            @Override
-            public List<Order> getAllOrdersForUser(String usersId) {
+    @Override
+    public List<Order> getAllOrdersForUser(String usersId) {
 
         return orderDAO.getAllByUserId(usersId);
-            }
+    }
 
-        @Override
-        public void confirmOrder (String orderId){
-            Order existingOrder = orderDAO.getOrderById(orderId).orElseThrow(() -> new IllegalArgumentException("Order not found"));
+    @Override
+    public void confirmOrder(String orderId) {
+        Order existingOrder = orderDAO.getOrderById(orderId).orElseThrow(() -> new IllegalArgumentException("Order not found"));
+
+        existingOrder.setStatus(CONFIRMED);
 
-            existingOrder.setStatus(CONFIRMED);
+        orderDAO.update(existingOrder);
+    }
 
-            orderDAO.update(existingOrder);
-        }
+    @Override
+    public List<Product> addProductToOrder(String name) {
+        productCart.add(productService.getProductByName(name));
+        return productCart;
+    };
 
-        @Override
-        public void checkoutOrder (Order order){
-         order.setStatus(PENDING);
-         orderDAO.save(order);
-        }
+    @Override
+    public void checkoutOrder() {
+        Order order = new Order();
+        order.setProductList(productCart);
+        order.setId(UUID.randomUUID().toString());
+        order.setStatus(PENDING);
+        order.setUserid(UserServiceImpl.getLoggedUser().get().getId());
```

Comment:
that's incorrect. U should use userId in a method signature checkoutOrder(String userId)
Also, you need to clear productCart. Because if user want to make another order, previous product list will be there